### DISCUSSION TITLE
a reload node already loaded while it should be zero load at reload node

### DISCUSSION THREAD

**Discussion by orangiee** [google/or-tools#2937] [Routing (and legacy CP) questions]
Hi, I'm using cvrp_reload.py, 

I got 50 nodes to delivery, and 5 vehicles, 5 reload nodes which correspondes to 5 vehicles. so I have
```py
 data['demands'] = [0, # depot 
-_capacity, # reload depot_first 1
-_capacity, # reload depot_second 2
-_capacity, # reload depot_third 3
-_capacity, # reload depot_fourth 4
-_capacity, # reload depot_fifth 5
 59,
 55,
 45,
 21,
 67,
...]
len(data['demands']) = 56
```
I printed out : vehicle id, start node of each vehicle, and end index. I found that start node and end index are beyond the size of data, for exemple, start nodes are: 57,58,59, 60, 61, and end index are 62, 63, 64, etc. The problem is, I got a result like this: 
`0 load:(0) -> 13 load(48) -> 2 load(67)`
question 1: why at node 13, it has load already 48 which should be zero since it begins with depot. 

I got also another thing: 
`0 load:(0) -> 14 load(28) -> 2 load(67) -> 23 load(17) ....  `
question 2:  why at node 23, it has also loaded which should be zero because reload node 2 has a negative capacity. 

Could anyone tell me the source of problem ? thank you so much in advance.

**Comment by orangiee** [comment]
Hi, could anyone answer this question, thanks a lot!

**Reply by Mizux** [reply]
1. You need to use `manager.IndexToNode()` internally solver duplicate them since it need unique id...
ps: also in your callback you use `manager.IndexToNode()` since solver will give you solver's space index as parameter

EDIT: I can see here, that this sample use the manager...
https://github.com/google/or-tools/blob/aa2d55f33dcadb9c3c5f97e06c1d313a58bd37de/ortools/constraint_solver/samples/cvrp_reload.py#L313-L314

**Reply by Mizux** [reply]
2. Please provide your code (or at least the link to your cvrp_reload.py ), CumulVar can't be negative, I suspect slackVar at node 2, from 2 -> 23 you have 67 -> 17 as load does `- vehicle_capacity == -50` ?

**Reply by orangiee** [reply]
Hi, thanks for your reply! 
1, I did exactly like the code: cvrp_reload.py, so I have the same data structure:
```python
data = {}
    _capacity = 15
    # Locations in block unit
    _locations = [
        (4, 4),  # depot
        (4, 4),  # unload depot_first
        (4, 4),  # unload depot_second
        (4, 4),  # unload depot_third
        (4, 4),  # unload depot_fourth
        (4, 4),  # unload depot_fifth
        (2, 0),
        (8, 0),  
        (0, 1),
        (1, 1),
        (5, 2),
        (7, 2),
        (3, 3),
        (6, 3),
        (5, 5),
        (8, 5),
        (1, 6),
        (2, 6),
        (3, 7),
        (6, 7),
        (0, 8),
        (7, 8)
    ]
data['demands'] = \
          [0, # depot
           -_capacity, # unload depot_first
           -_capacity, # unload depot_second
           -_capacity, # unload depot_third
           -_capacity, # unload depot_fourth
           -_capacity, # unload depot_fifth
           3, 3, # 1, 2
           3, 4, # 3, 4
           3, 4, # 5, 6
           8, 8, # 7, 8
           3, 3, # 9,10
           3, 3, # 11,12
           4, 4, # 13, 14
           8, 8] # 15, 16
```
But I added `capacity_dimension.CumulVar(node_index).RemoveInterval(0,600)` for vehicles. 
When I commented this line, the nodes after reload nodes are zero for their load and it's normal.  but when I uncommented this line, nodes after depot and after reload node have already loaded, I'm not sure if this cause the problem. 
ps: I found that at the end of each vehicle, their loads have the same value. I don't need that each vehicle have exactly the same value at the end of each route. what should I do ? 

2; I don't think that's the explanation, because I got other results which is not equal to -50. 

Thank you so much in advance for the help !
